packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -1231,28 +1233,20 @@ export class SideTabBar extends ScrollableTabBar {
                     if (iconElements.length === 1) {
                         const icon = iconElements[0];
                         rd.iconSize = { width: icon.clientWidth, height: icon.clientHeight };
-                        actualWidth += icon.clientHeight + paddingTop + paddingBottom + this.tabRowGap;
+                        actualWidth += icon.clientHeight + paddingTop + paddingBottom + marginTop;
 
-                        if (actualWidth > availableWidth && i !== 0) {
+                        if (actualWidth > availableWidth) {
                             rd.visible = false;
                             if (overflowStartIndex === -1) {
                                 overflowStartIndex = i;
                             }
                         }
+                        // Add the tab row gap to the actual width after the visibility check
+                        actualWidth += marginBottom + this.tabRowGap;
```

Comment:
Why does `paddingBottom` get added before the visibility check, but "marginBottom" afterwards?